### PR TITLE
EKF2 split output predictor and publish attitude immediately

### PR DESCRIPTION
**Pull request by dagar**
Requires https://github.com/PX4/ecl/pull/471

The EKF is updated to run the output predictor immediately after receiving new IMU data. Then PX4 side we publish vehicle_attitude immediately, and only publish positions at the EKF predication rate (every 8 milliseconds).

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@bkueng @RomanBapst Would be great if you could review & quickly flight-test.

**Comment by dagar** [comment]
This lowers EKF2 cpu usage considerably.

For example current EKF2 in master running a stock pixracer with active GPS fusion is ~14-16% cpu. With this PR it drops to around 10%.

File: src/modules/ekf2/ekf2_main.cpp [hunk 1]
```diff
@@ -1032,205 +1037,181 @@ void Ekf2::run()
...
-					_att_pub = orb_advertise(ORB_ID(vehicle_attitude), &att);
+		if (updated) {
 
-				} else {
-					orb_publish(ORB_ID(vehicle_attitude), _att_pub, &att);
+			filter_control_status_u control_status;
+			_ekf.get_control_mode(&control_status.value);
+
+			// only publish position after successful alignment
+			if (control_status.flags.tilt_align && control_status.flags.yaw_align) {
```
Comment (priseborough) [inline src/modules/ekf2/ekf2_main.cpp]: Should only be gated on tilt align. yaw alignment can momentarily go false.

Comment (dagar) [reply src/modules/ekf2/ekf2_main.cpp]: Makes sense, I'll change it. Currently it's both in master. https://github.com/PX4/ecl/blob/b26c2d62b80d6bee148fbfe01eb36809bfa721ea/EKF/ekf.cpp#L134

**Comment by priseborough** [comment]
Replay testing is showing https://github.com/PX4/ecl/pull/471 giving larger logged output observer tracking errors than ecl master so I will investigate further.

**Comment by priseborough** [comment]
@dagar has fixed the difference so it should now be good to flight test provided  we use https://github.com/PX4/ecl/pull/471 at commit  aaff047e5852761f8d15bc695c7a1ca70beeeeb1

**Comment by r0gelion** [comment]
@priseborough @dagar px4/ecl commit aaff047e5852761f8d15bc695c7a1ca70beeeeb1 hasn't been merged yet, do we need to wait for that commit to be merged or can we test this pull request as it is right now? we use to download the firmware from Jenkins artifacts.

**Comment by dagar** [comment]
Yes in this case it's okay to use the artifacts directly from Jenkins, but it was a good thing to check when other submodules are involved.

**Comment by santiago3dr** [comment]
a few flights on a pixracer (V4)
manual flight
https://logs.px4.io/plot_app?log=5920d2d1-36d8-47b1-918d-47bb85a45ef2
https://logs.px4.io/plot_app?log=4cc7676e-ae05-4f39-a64b-b55b39730d43

full auto from take-off to landing
https://logs.px4.io/plot_app?log=74ad4749-0a4b-4617-b2ee-c7e08194524f
https://logs.px4.io/plot_app?log=715517bf-e64d-493e-b954-1a7186fc297c

stable flights; didn't run into any issues

**Comment by dagar** [comment]
For reference here's a close comparison from master (same vehicle, same flight).
Master: https://logs.px4.io/plot_app?log=33cff20f-ef72-44ff-ba9d-440e26f352a0
PR: https://logs.px4.io/plot_app?log=74ad4749-0a4b-4617-b2ee-c7e08194524f

CPU usage before
image

CPU usage after
image

**Comment by Avysuarez** [comment]
Few flights with pixhawk 4 (V5).  Good flights, no issues.
https://review.px4.io/plot_app?log=0c36c73e-9881-4c6a-9cb6-bb95da46213a
https://review.px4.io/plot_app?log=fb23fb62-e7e5-42b9-82ad-577bd6bab31f
https://review.px4.io/plot_app?log=7c99b336-20f8-488f-9753-62d7aa1cd0b5

few flights with pixhawk 2.1 (v3). Good flights, no issues.
https://review.px4.io/plot_app?log=8054ec6d-142d-468b-bec0-cbf2d1d8f252
https://review.px4.io/plot_app?log=9d2ae4b6-1d66-4122-bccb-7a9152f25594
https://review.px4.io/plot_app?log=b632cd82-b9b5-4341-bb62-a6127e409f8a

**Comment by dannyfpv** [comment]
test flight = pixhawk mini good flight
https://logs.px4.io/plot_app?log=c5949d93-cbae-4841-8123-cbeade843571

pixhawk pro v4 good flight
https://logs.px4.io/plot_app?log=ce4bc8e9-c2ea-4288-bdc3-edcb2a89a1af

pixhack v5 great flight
https://logs.px4.io/plot_app?log=f5ae3251-e4fa-4e1b-bec8-0e9d7a5471da

**Pull request merged**